include/cuco/static_set.cuh

```diff
@@ -73,17 +73,17 @@ namespace experimental {
  * @throw If the probing scheme type is not inherited from `cuco::detail::probing_scheme_base`
  *
  * @tparam Key Type used for keys. Requires `cuco::is_bitwise_comparable_v<Key>`
- * @tparam Extent Data structure size type
  * @tparam Scope The scope in which operations will be performed by individual threads.
+ * @tparam Extent Data structure size type
  * @tparam KeyEqual Binary callable type used to compare two keys for equality
  * @tparam ProbingScheme Probing scheme (see `include/cuco/probing_scheme.cuh` for choices)
  * @tparam Allocator Type of allocator used for device storage
  * @tparam Storage Slot window storage type
  */
 
 template <class Key,
-          class Extent             = cuco::experimental::extent<std::size_t>,
           cuda::thread_scope Scope = cuda::thread_scope_device,
+          class Extent             = cuco::experimental::extent<std::size_t>,
```

Comment:
Since `cuda::thread_scope` is an enum type, for it to be available at compile time, we need to pass it in the template parameter list of the factory instead of the runtime argument list:
```cpp
make_static_set<Key>(cuda::thread_scope_device,, ...); // invalid; value is only known at runtime
make_static_set<Key, cuda::thread_scope_device>(...); // valid; value is known at compile time
```

That's why I needed to re-order these tparams.